### PR TITLE
Segtree: Monoid BitwiseOr/BitwiseAnd/BitwiseXor, get_slice

### PR DESCRIPTION
**Pull request by mizar**
- Monoid BitOrOper/BitAndOper/BitXorOper: for bool/bitset type segtree data
- FromIterator: struct from Iterator
- get_slice: allows slice access to internal data

usage sample: [ABC285F - Substring of Sorted String](https://atcoder.jp/contests/abc285/tasks/abc285_f)
https://atcoder.jp/contests/abc285/submissions/38317831

### CODE REVIEW

**Comment by mizar** [comment]
The following code keeps the segtree implementation, but changes the input/output processing.
https://atcoder.jp/contests/abc285/submissions/38161483

**Comment by mizar** [comment]
For the `FromIterator`, the following PR exists, so I'll just Revert it here
https://github.com/rust-lang-ja/ac-library-rs/pull/107

File: src/segtree.rs [hunk 1]
```diff
@@ -68,6 +68,48 @@ where
     }
 }
 
+pub struct BitOrOper<S>(Infallible, PhantomData<fn() -> S>);
```
Comment (TonalidadeHidrica) [inline src/segtree.rs:71]: This is just a matter of preference, but I prefer `BitwiseOr`.  ("Operation" is obvious in the context, while "bit or" is kinda misleading (are we handling a single bit?))

File: src/segtree.rs [hunk 2]
```diff
@@ -107,6 +149,10 @@ impl<M: Monoid> Segtree<M> {
         self.d[p + self.size].clone()
     }
 
+    pub fn get_slice(&self) -> &[M::S] {
+        &self.d[self.size..(self.size + self.n)]
```
Comment (TonalidadeHidrica) [inline src/segtree.rs:153]: Just showing off a trick: did you know that you can write like this 😜?  (I prefer this style becuase you need to specify the same variable twice.)

```suggestion
        &self.d[self.size..][..self.n]
```

File: src/segtree.rs [hunk 3]
```diff
@@ -68,6 +68,48 @@ where
...
 }
 
+pub struct BitOrOper<S>(Infallible, PhantomData<fn() -> S>);
+impl<S> Monoid for BitOrOper<S>
+where
+    S: Copy + BitOr<Output = S> + Default,
+{
+    type S = S;
+    fn identity() -> Self::S {
+        S::default()
```
Comment (TonalidadeHidrica) [inline src/segtree.rs:78]: I'm still doubting if we should use "Default" in this case, considering the semantics.  (See also my [previous comment](https://github.com/rust-lang-ja/ac-library-rs/pull/104#discussion_r1149274145)).

**Review by TonalidadeHidrica — APPROVED** [review]
LGTM

**Pull request merged**